### ISSUE TITLE
Use "german" instead of "ngerman" as polyglossia mainlanguage with lualatex

### ISSUE THREAD

**Issue by sscherfke** [sphinx-doc/sphinx#4717]
No german translations in PDFs generated by `make latexpdf` with `latex_engine = 'lualatex'`.

### Problem

When I have a Sphinx project with `language = 'german'` and `latex_engine ='lualatex'`, the generated PDF will have English names for the table of contents, chapter, figure and the like.

The reason for this is the `\setmainlanguage{ngerman}` command in the generated latex files.  If `german` was used instead, it would work as expected (you can add `r'\setmainlanguage{german}'` to `latex_elements['preamble']` to test it).

#### Procedure to reproduce the problem
```bash
$ sphinx-quickstart
Welcome to the Sphinx 1.7.1 quickstart utility.

...

For a list of supported codes, see
http://sphinx-doc.org/config.html#confval-language.
> Project language [en]: de

...

> Create Makefile? (y/n) [y]: 

...

Finished: An initial directory structure has been created.

...

$ vim conf.py
latex_engine = 'lualatex'
latex_elements = {
    # Use this to show that "german" works fine as language:
    # 'preamble': r'\setmainlanguage{german}',
}

$ make latexpdf
```

#### Error logs / results

PDF uses English terms (“contents” and “chapter” etc.)

#### Expected results

PDF uses German terms (“Inhaltsverzeichnis” and “Kapitel” etc.)

### Reproducible project / your project
Can be reproduces with a simple default project created by `sphinx-quickstart`.

### Environment info
- OS: Linux (Fedora 27)
- Python version: 3.6.4
- Sphinx version: 1.7.1
- texlive-luatex: Version svn40963, Release 36.fc27.5, Source texlive-2016-36.20160520.fc27.5.src.rpm
- texlive-polyglossia: Version svn40138

**Comment by sscherfke** [comment]
Using `german` seems not to work in a fedora26 docker image.  `hyph-utf8` throws an error there:
```
stack traceback:
        [C]: in function 'error'
        ...exlive/texmf-dist/tex/luatex/hyph-utf8/luatex-hyphen.lua:21: in function 'err'
        ...exlive/texmf-dist/tex/luatex/hyph-utf8/luatex-hyphen.lua:70: in function 'loadlanguage'
        ...texlive/texmf-dist/tex/latex/polyglossia/polyglossia.lua:41: in function 'loadlang'
        ...texlive/texmf-dist/tex/latex/polyglossia/polyglossia.lua:46: in function 'select_language'
        [\directlua]:1: in main chunk.
<argument> ...ing german}',\the \csname l@german\endcsname )}

l.20 \select@language {german}
```

I also wrote a minimal tex doc to see whether this might be an issue with polyglossia itself:
```tex
\documentclass[a4paper]{report}
\usepackage{polyglossia}
\setmainlanguage{ngerman}  % -> "Contents"
\setmainlanguage{german}  % -> "Inhaltsverzeichnis"

\begin{document}

\tableofcontents
\chapter{Hello, world!}
Ohai

\end{document}
```
I can indeed reproduce the issue with that.

**Comment by sscherfke** [comment]
Fedora 27 compiles with `german`, Fedora 26 does not.

**Comment by sscherfke** [comment]
The polyglossia docs indicate that `german` should be used.  Maybe it's a fedora26 bug with the hyphenation patterns when `german` is being used?

**Comment by tk0miya** [comment]
In my short investigation:

In babel, new orthography of German is represented as `ngerman`.
On the other hand, in polyglossia, it is represented as `spelling` option of `german` language:

>spelling = new (= 1996) or old (= 1901): indicates whether hyphenation
patterns for traditional (1901) or reformed (1996) orthography should be
used. The latter is the default.
>http://mirrors.rit.edu/CTAN/macros/latex/contrib/polyglossia/polyglossia.pdf

As a result, we need to switch the name of language when we use polyglossia.

Note: descriptions about new orthography:
* https://en.wikipedia.org/wiki/German_orthography_reform_of_1996
* https://tex.stackexchange.com/questions/67549/whats-the-difference-between-ngerman-and-german-in-babel

**Comment by tk0miya** [comment]
Fixed by #4876.
Thank you for reporting!
